src/script.c

```diff
@@ -541,34 +541,46 @@ int wally_scriptpubkey_p2sh_from_bytes(
     return ret;
 }
 
+int pubkey_compare(const void *a, const void *b)
+{
+    return memcmp(a, b, EC_PUBLIC_KEY_LEN);
+}
+
 int wally_scriptpubkey_multisig_from_bytes(
     const unsigned char *bytes, size_t bytes_len, uint32_t threshold,
     uint32_t flags, unsigned char *bytes_out, size_t len, size_t *written)
 {
     size_t n_pubkeys = bytes_len / EC_PUBLIC_KEY_LEN;
     size_t script_len = 3 + (n_pubkeys * (EC_PUBLIC_KEY_LEN + 1));
     size_t i;
+    unsigned char pubkey_bytes[bytes_len];
 
     if (written)
         *written = 0;
 
     if (!bytes || !bytes_len || bytes_len % EC_PUBLIC_KEY_LEN ||
         n_pubkeys < 1 || n_pubkeys > 16 || threshold < 1 || threshold > 16 ||
-        threshold > n_pubkeys || flags || !bytes_out || !written)
+        threshold > n_pubkeys || (flags & ~WALLY_SCRIPT_MULTISIG_SORTED) ||
+        !bytes_out || !written)
```

Comment:
!written may  be incorrect since we test it above and bypass written if is NULL

Reply:
I didn't touch that, but they way I see it, the _value_ of written is set to `0` if the _pointer_ exists. The check here cares if the pointer exists. It might make more sense to move the `*written = 0` below this line.

Reply:
you're right.  we can just leave it unless you want to change.